### DISCUSSION TITLE
iox::cxx::map implemenattion

### DISCUSSION THREAD

**Discussion by SomnathHolkar** [eclipse-iceoryx/iceoryx#1435] [General]
Iceoryx has almost all STL implemenation under hoofs i.e. all datatypes are supported by iceoryx hoofs except **map**

only std::map which actually usage the dynamic memory internally needs to be added in hoofs.

**Comment by elfenpiff** [comment]
@SomnathHolkar you are right! When you take a glimpse at my inofficial private repo: https://gitlab.com/elfenpiff/iceoryx_on_fire/-/tree/master/include you see that I started this some weeks ago but didn't find the time to continue this.

To provide you an update what we still need:
1. Red-Black-Tree implementation, is more or less finished but untested
2. Based on the Red-Black-Tree we can implement a `cxx::map` and a `cxx::set`.

But if you want you are invited to take over this task. We would very much welcome it and support you along the way!

**Reply by hemalbavishi** [reply]
@elfenpiff : We tried enhancing your part but faced few issues. Do you have some time next week to discuss on that?

Best Regards,
Hemal Bavishi

**Reply by elfenpiff** [reply]
@hemalbavishi I would be happy to help!

Please suggest a date. I am available the whole week except on:
* 17.01.2023 04:30pm - 05:30pm (CET)
* 18.01.2023 10:00am - 01:15pm (CET)

Would you like to meet via gmeet?
